src/modules/network/implementation/libp2p-service.js

```diff
@@ -245,11 +245,25 @@ class Libp2pService {
                 this.logger.debug(
                     `Receiving message from ${peerIdString} to ${this.config.id}: protocol: ${protocol}, messageType: ${message.header.messageType};`,
                 );
-                await handler(message, peerIdString);
+                const modifiedMessage = this.modifyMessage(message, blockchainImplementations);
+                await handler(modifiedMessage, peerIdString);
             }
         });
     }
 
+    modifyMessage(message, blockchainImplementations) {
```

Comment:
this should probably not be in the libp2p service as it is modifying the message data and is "aware" of its contents

Reply:
++